tests/components/wled/test_light.py

```diff
@@ -137,6 +142,35 @@ async def test_switch_change_state(
         )
 
 
+async def test_dynamically_handle_segments(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test if a new/deleted segment is dynamically added/removed."""
+    await init_integration(hass, aioclient_mock)
+
+    assert hass.states.get("light.wled_rgb_light")
+    assert hass.states.get("light.wled_rgb_light_1")
+
+    data = json.loads(load_fixture("wled/rgb_single_segment.json"))
+    device = WLEDDevice(data)
+
+    # Test removal if segment went missing
+    with patch(
+        "homeassistant.components.wled.WLED.update", return_value=device,
+    ):
+        async_fire_time_changed(hass, dt_util.utcnow() + SCAN_INTERVAL)
+        await hass.async_block_till_done()
+        assert hass.states.get("light.wled_rgb_light")
+        assert not hass.states.get("light.wled_rgb_light_1")
+
+    # Test adding if segment shows up again
+    async_fire_time_changed(hass, dt_util.utcnow() + SCAN_INTERVAL)
```

Comment:
Do we need to change time from the last update?

Reply:
Not really, as it just fires the event again. Effectively, we are not changing the current time here, just faking the fired time.

Reply:
Ok. I was thinking about the coordinator debouncer and if that would interfere with the second call.

Reply:
Yes, that was my initial hunch as well. However, as the previous is blocked until done, the coordinator is free when we reach this point.

Reply:
There's a default cooldown of 10 seconds. We're not blocking for that time, right?

Reply:
Nevermind, the coordinator refresh on interval cancels the timer task of the debouncer. It's only `coordinator.async_request_refresh` that is debounced.